src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -417,15 +423,39 @@ void GcsServer::StoreGcsServerAddressInRedis() {
 }
 
 void GcsServer::InitRaySyncer(const GcsInitData &gcs_init_data) {
-  /*
-    The current synchronization flow is:
-        raylet -> syncer::poller --> syncer::update -> gcs_resource_manager
-        gcs_placement_scheduler --/
-  */
-  ray_syncer_ = std::make_unique<gcs_syncer::RaySyncer>(
-      main_service_, raylet_client_pool_, *gcs_resource_manager_);
-  ray_syncer_->Initialize(gcs_init_data);
-  ray_syncer_->Start();
+  if (RayConfig::instance().use_ray_syncer()) {
+    ray_syncer_node_id_ = NodeID::FromRandom();
+    ray_syncer_ = std::make_unique<syncer::RaySyncer>(ray_syncer_io_context_,
+                                                      ray_syncer_node_id_.Binary());
+    ray_syncer_->Register(
+        syncer::RayComponentId::RESOURCE_MANAGER, nullptr, gcs_resource_manager_.get());
+    ray_syncer_->Register(
+        syncer::RayComponentId::SCHEDULER, nullptr, gcs_resource_manager_.get());
+    ray_syncer_thread_ = std::make_unique<std::thread>([this]() {
+      boost::asio::io_service::work work(ray_syncer_io_context_);
+      ray_syncer_io_context_.run();
+    });
+
+    for (const auto &pair : gcs_init_data.Nodes()) {
```

Comment:
The dead node should be filtered out.

Reply:
I'm actually following the old pattern and it doesn't check it as well.
I'll add the check here.